frodo2-runtime-android/build.gradle

```diff
@@ -48,68 +25,73 @@ android {
-          break;
-        case IMessage.INFO:
-          log.info message.message, message.thrown
-          break;
-        case IMessage.DEBUG:
-          log.debug message.message, message.thrown
-          break;
-      }
-    }
-  }
-
-  //Change aar file name: add version name only for debug version (development)
-  if (variant.name == 'debug') {
-    variant.outputs.each { output ->
-      if (output.outputFile != null && output.outputFile.name.endsWith('.aar')) {
-        def file = output.outputFile
-        def fileName = file.name.replace("${variant.buildType.name}.aar", "${project.version}.aar")
-        output.outputFile = new File(file.parent, fileName)
-      }
-    }
+    targetCompatibility JavaVersion.VERSION_1_8
+    sourceCompatibility JavaVersion.VERSION_1_8
   }
 }
 
+// TODO: Do we need this?
+//android.libraryVariants.all { variant ->
+//  LibraryPlugin plugin = project.plugins.getPlugin(LibraryPlugin)
+
+//Compile aspects
+//  JavaCompile javaCompile = variant.javaCompile
+//  javaCompile.doLast {
+//    String[] args = ["-showWeaveInfo",
+//                     "-1.5",
+//                     "-inpath", javaCompile.destinationDir.toString(),
+//                     "-aspectpath", javaCompile.classpath.asPath,
+//                     "-d", javaCompile.destinationDir.toString(),
+//                     "-classpath", javaCompile.classpath.asPath,
+//                     "-bootclasspath", plugin.project.android.bootClasspath.join(File.pathSeparator)]
+
+//    final MessageHandler handler = new MessageHandler(true);
+//    final def log = project.logger
+//    new Main().run(args, handler)
+//    for (IMessage message : handler.getMessages(null, true)) {
+//      switch (message.getKind()) {
+//        case IMessage.ABORT:
+//        case IMessage.ERROR:
+//        case IMessage.FAIL:
+//          log.error message.message, message.thrown
+//          break;
+//        case IMessage.INFO:
+//          log.info message.message, message.thrown
+//          break;
+//        case IMessage.DEBUG:
+//          log.debug message.message, message.thrown
+//          break;
+//      }
+//    }
+//  }
+
+// TODO: This need to be upgraded for gradle tools 3.0.0. Looking at the current output filename
+// TODO: I get frodo2-runtime-android-[version].aar which matched the java equivalent of
+// TODO: frodo2-runtime-java-[version].jar
+//Change aar file name: add version name only for debug version (development)
+//  if (variant.name == 'debug') {
+//    variant.outputs.each { output ->
+//      if (output.outputFile != null && output.outputFile.name.endsWith('.aar')) {
+//        def file = output.outputFile
+//        def fileName = file.name.replace("${variant.buildType.name}.aar", "${project.version}.aar")
+//        output.outputFile = new File(file.parent, fileName)
+//      }
+//    }
+//  }
+//}
+
+// TODO: Why is this needed? The artifact seems to be published okay on maven local without this
```

Comment:
As soon as the android artifact keeps this format and name, everything is fine:
```frodo2-runtime-android-${project.version}.aar```

I had problems in the past to make everything works. Because of all the moving parts of this project. Maybe back with your updates this works out of the box. 

Whenever you wanna try things out...just make sure that the examples work. First wipe out everything frodo2 related in your local maven repository and executes frodo2 installation plus examples:

```
./install_frodo2.sh
./run_frodo2_java
./run_frodo2_android
```
You should see on the default output observables being logged.